javascript/index.js

```diff
@@ -4,37 +4,144 @@
+              );
+            },
+            (error) => console.log(error)
+          );
+        },
+        (error) => console.log(error)
+      );
+    },
+    (error) => console.log(error)
+  );
 
 // Iteration 2 - using promises
-// ...
+
+obtainInstruction("steak", 0).then((step0) => {
+  document.querySelector("#steak").innerHTML += `<li>${step0}</li>`;
+  obtainInstruction("steak", 1).then((step1) => {
+    document.querySelector("#steak").innerHTML += `<li>${step1}</li>`;
+    obtainInstruction("steak", 2).then((step2) => {
+      document.querySelector("#steak").innerHTML += `<li>${step2}</li>`;
+      obtainInstruction("steak", 3).then((step3) => {
+        document.querySelector("#steak").innerHTML += `<li>${step3}</li>`;
+        obtainInstruction("steak", 4).then((step4) => {
+          document.querySelector("#steak").innerHTML += `<li>${step4}</li>`;
+          obtainInstruction("steak", 5).then((step5) => {
+            document.querySelector("#steak").innerHTML += `<li>${step5}</li>`;
+            obtainInstruction("steak", 6).then((step6) => {
+              document.querySelector("#steak").innerHTML += `<li>${step6}</li>`;
+              obtainInstruction("steak", 7)
+                .then((step7) => {
+                  document.querySelector(
+                    "#steak"
+                  ).innerHTML += `<li>${step7}</li>`;
+                  document.querySelector(
+                    "#steak"
+                  ).innerHTML += `<li>Stake is ready!</li>`;
+                })
+                .then(() =>
+                  document.querySelector("#steakImg").removeAttribute("hidden")
+                );
+            });
+          });
+        });
+      });
+    });
+  });
+});
 
 // Iteration 3 using async/await
-// ...
 
+const asyncAndAwaitReturn = async () => {
+  try {
+    let step0 = await obtainInstruction("broccoli", 0);
+    document.querySelector("#broccoli").innerHTML += `<li>${step0}</li>`;
+    let step1 = await obtainInstruction("broccoli", 1);
+    document.querySelector("#broccoli").innerHTML += `<li>${step1}</li>`;
+    let step2 = await obtainInstruction("broccoli", 2);
+    document.querySelector("#broccoli").innerHTML += `<li>${step2}</li>`;
+    let step3 = await obtainInstruction("broccoli", 3);
+    document.querySelector("#broccoli").innerHTML += `<li>${step3}</li>`;
+    let step4 = await obtainInstruction("broccoli", 4);
+    document.querySelector("#broccoli").innerHTML += `<li>${step4}</li>`;
+    let step5 = await obtainInstruction("broccoli", 5);
+    document.querySelector("#broccoli").innerHTML += `<li>${step5}</li>`;
+    let step6 = await obtainInstruction("broccoli", 6);
+    document.querySelector("#broccoli").innerHTML += `<li>${step6}</li>`;
+    document.querySelector("#broccoliImg").removeAttribute("hidden");
+  } catch (err) {
+    console.log(err);
+  }
+};
+asyncAndAwaitReturn();
 // Bonus 2 - Promise all
-// ...
\ No newline at end of file
+// const tst =
+Promise.all([
```

Comment:
Everything is working like it should be, but you did the broccoli again instead of the brussels sprouts